Code/Framework/AzCore/AzCore/std/parallel/binary_semaphore.h

```diff
@@ -38,13 +38,13 @@ namespace AZStd
 
         binary_semaphore(bool initialState = false)
         {
-            m_event = CreateEvent(nullptr, false, initialState, nullptr);
+            m_event = CreateEventW(nullptr, false, initialState, nullptr);
```

Comment:
is CreateEventW a windows function? It looks like a windows api style name.

If it is should this be PALified?

Reply:
It is, this is somehow "palified" but it was done with `AZ_TRAIT_OS_USE_WINDOWS_SET_EVENT`, seems like it should be  done with implementations in different headers instead, but there are 3 versions, two of them Linux with a max count of sempahores
I leave it outside of the scope of this PR... (already taking too much here)